### DISCUSSION TITLE
WARNING:py.warnings:/usr/local/lib/python3.10/site-packages/urllib3/connectionpool.py:1013: InsecureRequestWarning: Unverified HTTPS request is being made to host "<elastic-host>"

### DISCUSSION THREAD

**Discussion by alexku7** [jertel/elastalert2#574] [Q&A] [answered]
The elastalert supports the "ssl_show_warn" flag to suppress warnings when it sends alerts to various targets but , as far as I can understand the elastalert doesn't support this flag when it polls the elasticsearch .

In many cases the elastalert2 is deployed ont he same cluster with ELK and polls the elastic via kubernetes internal names, therefore can't verify the TLS certificates. As result the log is blown with InsecureRequestWarning warnings.

**Comment by nsano-rururu** [comment, marked as answer]
It is an option when connecting elasticsearch, not a polling option.

**Reply by alexku7** [reply]
Maybe I didn't describe it correctly. I meant "polling=connecting".

The bottom line,  in my scenario, the elastalert is installed on the same kubernetes with the ELK. 
The elastalert connects to the ELK by kubernetes internal name but over HTTPS. Since the certificate doesn't match the internal kubernetes name, the elastalert2 prints many useless warning to the log.


Could be god to add an ability to suppress these warnings

**Reply by nsano-rururu** [reply]
elastalert2 uses elasticsearch-py 7.0.0, but since the following modifications of ssl_show_warn have been made after elasticsearch-py 7.0.0, use the ssl_show_warn option to disable InsecureRequestWarning from the urllib3 connection pool. I don't think it's supported.

>ssl_show_warn option improvements
https://github.com/elastic/elasticsearch-py/pull/1076

**Reply by nsano-rururu** [reply]
elastalert2 cannot be upgraded because it has a stuck feature that can be upgraded to versions above elasticsearch-py 7.0.0. Please understand.

**Reply by jertel** [reply]
`ssl_show_warn` is already being passed into the official elasticsearch-py library, as documented [here](https://elasticsearch-py.readthedocs.io/en/7.x/transports.html#urllib3httpconnection) and implemented in [elastalert/__init__.py](https://github.com/jertel/elastalert2/blob/c28130debacf023406fce7dbaf60d1f68e540af7/elastalert/__init__.py#L26).

I just ran a test against a self-signed ES server with an altered subject name, and I'm not seeing the Insecure warning being spammed. I'm using the `jertel/elastalert2` Docker image. Since I'm unable to reproduce it you may need to debug this on your own to see if there's a problem in ElastAlert 2,a problem in the elasticsearch-py lib, or perhaps even in the Python `urllib3` library.

**Reply by jertel** [reply]
Thanks for pointing out that link @nsano-rururu. It's possible that my self-signed cert is interfering with my local test, which could explain why I wasn't able to reproduce it.

@alexku7 See #90 for the details behind the pin to 7.0.0, with further discussion in #92.

**Reply by nsano-rururu** [reply]
@jertel 

ssl_show_warn is also elasticsearch-py 7.0.0. After the 7.0.0 release, we have confirmed that the ssl_show_warn option is used to disable InsecureRequestWarning from the urllib3 connection pool. I tried the source code.
The following support is not included in elasticsearch-py 7.0.0.

>ssl_show_warn option improvements
https://github.com/elastic/elasticsearch-py/pull/1076

**Reply by nsano-rururu** [reply]
@jertel 

Please complain after looking at the source code properly. ask.

**Reply by alexku7** [reply]
Ok, I see the problem.

Thank you for assistance.
Probably currently we will try to ignore these warnings in the logs :)

**Reply by nsano-rururu** [reply]
It seems that the correspondence has been merged in elasticserarch-py 7.5.0.

**Reply by nsano-rururu** [reply]
In the future, if we end support for elasticsearch in the past and raise elasticserarch-py to 7.5.0 or later, I think that we will be able to do what we want to do this time. We apologize for any inconvenience.

**Reply by nsano-rururu** [reply]
The reason why `elasticsearch> = 7.5.0',` was included in the pull request was connected.
I didn't notice it at the time of the review.

Task/show warn false
https://github.com/jertel/elastalert2/pull/87

**Reply by nsano-rururu** [reply]
The person who submitted the ssl_show_warn pull request to elasticsearch-py and the person who submitted the ssl_show_warn pull request to elastalert2 were the same person. just noticed.
Same person
